atip/simulator.py

```diff
@@ -186,35 +198,55 @@ def _recalculate_phys_data(self, callback):
                     )
                 except Exception as e:
                     warn(at.AtWarning(e))
-                # Signal up to date before the callback is executed in case
-                # the callback requires data that requires the calculation
-                # to be up to date.
+                # Signal the up to date flag since the physics data is now up to date.
+                # We do this before the callback is executed in case the callback
+                # checks the flag.
                 self.up_to_date.Signal()
                 if callback is not None:
                     logging.debug("Executing callback function.")
                     callback()
                     logging.debug("Callback completed.")
+            # This could cause thread-locking issues if another application using
+            # cothreads is poorly written and running on the same machine, but this
```

Comment:
So a completely separate application running on the same machine can cause problems here?

Or do you mean on the same Python process?

Reply:
Yes, separate application, doesn't have to be the same Python process